### PR TITLE
Add option image_type in cluster_autoscaling block

### LINKED ISSUE #1191: No Option for specifying image_type in auto_provisioning_defaults
### TL;DR

The `auto_provisioning_defaults` block supports [image_type](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#image_type)

But the option is not available currently in the module.

Here is how it could look like:

```
cluster_autoscaling {
    enabled = var.cluster_autoscaling.enabled
    dynamic "auto_provisioning_defaults" {
      for_each = var.cluster_autoscaling.enabled ? [1] : []

      content {
        service_account   = local.service_account
        oauth_scopes       = local.node_pools_oauth_scopes["all"]
        min_cpu_platform = lookup(var.node_pools[0], "min_cpu_platform", "")
        image_type           = lookup(var.node_pool[0], "image_type", "")
      }
    }

```

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#image_type
```

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by tweiss-mdm**
Fixes #1191

### CODE REVIEW

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -111,6 +111,7 @@ resource "google_container_cluster" "primary" {
         {% if beta_cluster %}
         min_cpu_platform = lookup(var.node_pools[0], "min_cpu_platform", "")
         {% endif %}
+        image_type = lookup(var.node_pools[0], "image_type", "")
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:114]: Looks like there is a validation that prevents empty string as default. Can we try null instead?
https://github.com/hashicorp/terraform-provider-google/blob/627220d3985030c1a65c87faa9ea882e46dc0e63/google/resource_container_cluster.go#L341

**Review by bharathkkb — CHANGES_REQUESTED** [review]

File: autogen/main/cluster.tf.tmpl (hunk 1 above)
Comment (tweiss-mdm) [reply autogen/main/cluster.tf.tmpl]: null is not accepted either, trying with the default value "COS_CONTAINERD"

Comment (bharathkkb) [reply autogen/main/cluster.tf.tmpl]: @tweiss-mdm thats reasonable

**Review by bharathkkb — COMMENTED** [review]
@tweiss-mdm Can you also sign the CLA
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1192/checks?check_run_id=5787910422

**Pull request closed without merging**